### PR TITLE
Bleeding overhaul

### PR DESCRIPTION
**Pull request by Crystalwarrior**
_for every small feature i make i have to make like 20 fucking overhauls i swear to god what is this_
This PR overhauls bleeding in terms of it's use as well as the recovery methods.
Before, bleeding was determined by raw brute variables on all of your limbs. Now it's determined by a special "bloodloss" variable, which goes from 0 to 2, that determines amount of units of blood you lose per limb. This means that SHARP weapons are more likely to cause bloodsplatters and bloodloss than BLUNT weapons. I also make use of SHARP_ACCURATE flag for knives and scalpels to make them even more viable for causing bloodloss.

Bleeding chance is determined by this algorithm:
`prob(ITEM.force * max(ITEM.sharpness+1, 1) + LIMB.brute_dam/2)`
where prob is probability from 0 to 100, ITEM.sharpness is a variable from 0 to 2 (0 is BLUNT, 1 is SHARP, 2 is SHARP_ACCURATE) and LIMB.brute_dam is the limb's brute damage.
To put it simply, it checks the item sharpness, item force and affected limb's existing brute damage for the probability. All these factors will increase the chances of causing a blood splatter and bloodloss.
ALSO, this PR makes blood splatters a good visual indicator that you have caused your victim to bleed.

Note that this is not an accurate probability since prior to that there's also probability for bloodifying the item itself:
`if(ITEM.force && prob(25 + (ITEM.force * 2))) //45% on toolboxes (10 force)`

and here's a fukken changelog copypasta because I'm lazy.
- experiment: "Bloodloss has now become a seperate damage variable on organs as opposed to something determined by brute damage! This means that sharp weapons are more likely to cause bleeding than blunt."
- tweak: "Most knives have been given an \"IS_SHARP_ACCURATE\" tag, which lets you use these knives to cut food + it has an additional bleeding buff. Scalpel also has this tag, for the record."
- tweak: "Medical Analyzers interface has been given a slight overhaul. Damage reports are now properly tabulated, bleed variable is now visible on limbs. It determines the amount of blood lost every tick (2 seconds). Max bloodloss for every individual limb is 2."
- tweak: "Styptic Patches now completely heal bloodloss. Bruise packs remove 1 point of bloodloss."
- tweak: "Buffed iron reagent to restore 0.6u of blood every tick instead of 0.4u. Nutriment still restores 0.4u. What this means is that if you use iron + eat something you'll restore 1u of blood every tick (2 seconds)."
- tweak: "Most injected reagents and a bunch of other items (like mediguns/bibles) heal a small bit of bloodloss as well."
- tweak: "IV drips have been sped up slightly to restore 15u of blood instead of 10u every tick"
- tweak: "If your limb has 0 brute damage it will no longer have bloodloss"
- tweak: "Cautery and Welding Tool both can close up bleeding wounds. Completely healing the brute damage on a limb will also remove bloodloss."
- tweak: "Embedded objects now cause bloodloss on limb instead of being hard-checked by blood code."
- rscadd: "Surgery freezer now has 2 of each blood type + 3 random bloodtype bloodpacks. Empty bloodpacks are removed because nobody will fuckin donate blood on this server."

### CODE REVIEW

**Comment by Carbonhell** [comment]
i'm legitly scared please review your own code 5 times before asking me to codecheck it, i don't want this to mess up anyhow on live otherwise people are seriously gonna lynch you

**Comment by Spacedong** [comment]
his changes are so big people wiill lynch him anyway, just let him die for the cause

**Comment by Crystalwarrior** [comment]
This PR is not merge-ready yet don't worry catbon

**Comment by Crystalwarrior** [comment]
It's almost merge-ready now, tested most of this shit on my local server. All that remains is a codecheck and head input.

**Comment by KayeArray** [comment]
Sounds good.

**Comment by Crystalwarrior** [comment]
Still needs head input

**Comment by doobiedoo23** [comment]
:+1:

**Comment by Nokington** [comment]
Sure

**Comment by Lawligagger** [comment]
+1

**Comment by Carbonhell** [comment]
will codecheck tomorrow

**Comment by calibraptor** [comment]
Sounds good to me

File: code/game/machinery/pooldrain.dm [hunk 1]
```diff
@@ -72,7 +72,7 @@
 						if(whirlm.health <= -50) //If very damaged, gib.
 							whirlm.gib()
 						if(whirlm.stat != CONSCIOUS || whirlm.lying) // If
-							whirlm.adjustBruteLoss(-5)
```
Comment (Carbonhell) [inline code/game/machinery/pooldrain.dm:75]: Wat,the whirl healed you Before? Lmao

File: code/game/objects/items/weapons/tools.dm [hunk 2]
```diff
@@ -257,17 +257,24 @@
...
 
-	if(affecting.status == ORGAN_ROBOTIC && user.a_intent != "harm")
-		if(src.remove_fuel(1))
-			playsound(loc, 'sound/items/Welder.ogg', 50, 1)
-			user.visible_message("<span class='notice'>[user] starts to fix some of the dents on [H]'s [affecting.getDisplayName()].</span>", "<span class='notice'>You start fixing some of the dents on [H]'s [affecting.getDisplayName()].</span>")
-			if(!do_mob(user, H, 50))	return
-			item_heal_robotic(H, user, 5, 0)
+	if(user.a_intent != "harm")
+		if(affecting.status == ORGAN_ORGANIC && affecting.bloodloss > 0)
+			if(src.remove_fuel(1))
```
Comment (Carbonhell) [inline code/game/objects/items/weapons/tools.dm]: Sssrrrccccccc

File: code/game/objects/items/weapons/tools.dm [hunk 3]
```diff
@@ -257,17 +257,24 @@
...
+				if(!do_mob(user, H, 50))	return
+				user.visible_message("<span class='notice'>[user] has closed up wounds [H]'s [affecting.getDisplayName()].</span>", "<span class='notice'>You closed up wounds on [H]'s [affecting.getDisplayName()].</span>")
+				affecting.heal_damage(bleed=affecting.bloodloss)
+				affecting.take_damage(burn=10) //Quite harsh tradeoff
 			return
-		else
-			return
-	else
-		return ..()
+		else if(affecting.status == ORGAN_ROBOTIC && affecting.get_damage() > 0)
```
Comment (Carbonhell) [inline code/game/objects/items/weapons/tools.dm:270]: Is get damage the right proc?you only want to fix brute DMG, burn gets fixed by wires

File: code/modules/mob/living/carbon/human/blood.dm [hunk 4]
```diff
@@ -108,18 +108,22 @@ var/const/BLOOD_VOLUME_SURVIVE = 122
...
 			//We want an accurate reading of .len
-			listclearnulls(org.embedded_objects)
-			blood_max += 0.5*org.embedded_objects.len
-
-			if(brutedamage > 30)
-				blood_max += 0.5
-			if(brutedamage > 50)
-				blood_max += 1
-			if(brutedamage > 70)
-				blood_max += 2
```
Comment (Carbonhell) [inline code/modules/mob/living/carbon/human/blood.dm:122]: Delete this commwnted out code

File: code/modules/mob/living/carbon/human/human_damage.dm [hunk 5]
```diff
@@ -45,6 +50,12 @@
 	else
 		heal_overall_damage(0, -amount)
 
+/mob/living/carbon/human/adjustBloodLoss(amount)
+	if(amount > 0)
```
Comment (Carbonhell) [inline code/modules/mob/living/carbon/human/human_damage.dm:54]: Why this if? Isn't it useless

File: code/modules/mob/living/carbon/human/human_damage.dm [hunk 6]
```diff
@@ -45,6 +50,12 @@
...
+		take_overall_damage(0, 0, amount)
+	else
+		heal_overall_damage(0, 0, -amount)
```
Comment (Carbonhell) [inline code/modules/mob/living/carbon/human/human_damage.dm:57]: Oh nvm i see,all good

File: code/modules/surgery/generic_steps.dm [hunk 7]
```diff
@@ -67,6 +66,9 @@
 /datum/surgery_step/close/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
 	if(locate(/datum/surgery_step/saw) in surgery.steps)
 		target.heal_organ_damage(45,0)
+	if(istype(surgery.organ, /obj/item/organ/limb))
+		var/obj/item/organ/limb/L
+		L.heal_damage(bleed=L.bloodloss) //you mend the incision, don't you?
```
Comment (Carbonhell) [inline code/modules/surgery/generic_steps.dm:71]: Isn't it hemostat job to stop bleeding

**Comment by Carbonhell** [comment]
Fix what i pointed out and it'll be ok

**Comment by myontasm** [comment]
My only concern is that this might invalidate bandages completely.

**Comment by KayeArray** [comment]
Bandages are already useless because healing the brute stops bleeding anyway.

**Comment by ghost** [comment]
Nah, bandages in medbay are useless for the most part but carrying some gauze made from bedsheets can save ya

File: code/modules/surgery/generic_steps.dm (hunk 7 above)
Comment (Crystalwarrior) [inline code/modules/surgery/generic_steps.dm:71]: yes, and now that's fixed

File: code/modules/mob/living/carbon/human/human_damage.dm (hunk 5 above)
Comment (Crystalwarrior) [inline code/modules/mob/living/carbon/human/human_damage.dm:54]: wtf, no it isn't, also this adjustBloodLoss function is the same as adjustBruteLoss, etc.

File: code/modules/mob/living/carbon/human/blood.dm (hunk 4 above)
Comment (Crystalwarrior) [inline code/modules/mob/living/carbon/human/blood.dm:122]: already done

File: code/game/objects/items/weapons/tools.dm (hunk 3 above)
Comment (Crystalwarrior) [inline code/game/objects/items/weapons/tools.dm:270]: yes. yes it is.

File: code/game/objects/items/weapons/tools.dm (hunk 2 above)
Comment (Crystalwarrior) [inline code/game/objects/items/weapons/tools.dm]: fkkoffffff

File: code/game/machinery/pooldrain.dm (hunk 1 above)
Comment (Crystalwarrior) [inline code/game/machinery/pooldrain.dm:75]: yup

**Comment by Crystalwarrior** [comment]
Imo Bandages should patch up the "bleeding" on the wound completely instead of being a time-delay. Though that means they will become limb-based.

File: code/game/objects/items/weapons/tools.dm [hunk 8]
```diff
@@ -257,17 +257,24 @@
...
-			playsound(loc, 'sound/items/Welder.ogg', 50, 1)
-			user.visible_message("<span class='notice'>[user] starts to fix some of the dents on [H]'s [affecting.getDisplayName()].</span>", "<span class='notice'>You start fixing some of the dents on [H]'s [affecting.getDisplayName()].</span>")
-			if(!do_mob(user, H, 50))	return
-			item_heal_robotic(H, user, 5, 0)
+	if(user.a_intent != "harm")
+		if(affecting.status == ORGAN_ORGANIC && affecting.bloodloss > 0)
+			if(src.remove_fuel(1))
+				playsound(loc, 'sound/items/Welder.ogg', 50, 1)
+				user.visible_message("<span class='notice'>[user] starts to close up wounds on [H]'s [affecting.getDisplayName()].</span>", "<span class='notice'>You start closing up wounds on [H]'s [affecting.getDisplayName()].</span>")
+				if(!do_mob(user, H, 50))	return
```
Comment (Carbonhell) [inline code/game/objects/items/weapons/tools.dm:265]: you didn't space return

File: code/game/objects/items/weapons/tools.dm [hunk 9]
```diff
@@ -257,17 +257,24 @@
...
-		else
-			return
-	else
-		return ..()
+		else if(affecting.status == ORGAN_ROBOTIC && affecting.get_damage() > 0)
+			if(src.remove_fuel(1))
+				playsound(loc, 'sound/items/Welder.ogg', 50, 1)
+				user.visible_message("<span class='notice'>[user] starts to fix some of the dents on [H]'s [affecting.getDisplayName()].</span>", "<span class='notice'>You start fixing some of the dents on [H]'s [affecting.getDisplayName()].</span>")
+				if(!do_mob(user, H, 50))	return
+				item_heal_robotic(H, user, 5, 0)
```
Comment (Carbonhell) [inline code/game/objects/items/weapons/tools.dm:275]: return missing a space

File: code/game/objects/items/weapons/tools.dm [hunk 10]
```diff
@@ -257,17 +257,24 @@
...
+				user.visible_message("<span class='notice'>[user] has closed up wounds [H]'s [affecting.getDisplayName()].</span>", "<span class='notice'>You closed up wounds on [H]'s [affecting.getDisplayName()].</span>")
+				affecting.heal_damage(bleed=affecting.bloodloss)
+				affecting.take_damage(burn=10) //Quite harsh tradeoff
 			return
-		else
-			return
-	else
-		return ..()
+		else if(affecting.status == ORGAN_ROBOTIC && affecting.get_damage() > 0)
+			if(src.remove_fuel(1))
```
Comment (Carbonhell) [inline code/game/objects/items/weapons/tools.dm]: src

File: code/modules/mob/living/carbon/human/blood.dm [hunk 11]
```diff
@@ -108,18 +108,22 @@ var/const/BLOOD_VOLUME_SURVIVE = 122
 		//Bleeding out
 		blood_max = 0
 		for(var/obj/item/organ/limb/org in organs)
-			var/brutedamage = org.brute_dam
+			// var/brutedamage = org.brute_dam
```
Comment (Carbonhell) [inline code/modules/mob/living/carbon/human/blood.dm]: delete

File: code/modules/mob/living/carbon/human/blood.dm [hunk 12]
```diff
@@ -108,18 +108,22 @@ var/const/BLOOD_VOLUME_SURVIVE = 122
...
-			blood_max += 0.5*org.embedded_objects.len
-
-			if(brutedamage > 30)
-				blood_max += 0.5
-			if(brutedamage > 50)
-				blood_max += 1
-			if(brutedamage > 70)
-				blood_max += 2
+			// listclearnulls(org.embedded_objects)
+			// blood_max += 0.5*org.embedded_objects.len //Let's instead add bloodloss where pain is handled.
```
Comment (Carbonhell) [inline code/modules/mob/living/carbon/human/blood.dm]: wtf you didn't

File: code/modules/surgery/tools.dm [hunk 13]
```diff
@@ -32,6 +32,20 @@
...
 
+/obj/item/weapon/cautery/attack(mob/living/carbon/human/H, mob/user)
+	if(!istype(H))
+		return ..()
+
+	var/obj/item/organ/limb/affecting = H.get_organ(check_zone(user.zone_sel.selecting))
+
+	if(user.a_intent != "harm" && affecting.status == ORGAN_ORGANIC && affecting.bloodloss > 0)
+		user.visible_message("<span class='notice'>[user] starts to close up wounds on [H]'s [affecting.getDisplayName()].</span>", "<span class='notice'>You start closing up wounds on [H]'s [affecting.getDisplayName()].</span>")
+		if(!do_mob(user, H, 30))	return
```
Comment (Carbonhell) [inline code/modules/surgery/tools.dm]: spacing

**Comment by Crystalwarrior** [comment]
@Carbonhell codecheck ok??

**Comment by blobbernaut** [comment]
tweak: "Embedded objects now cause bloodloss on limb instead of being hard-checked by blood code."

so you can wrap up an embed with gauze to stop the bleeding? What's the purpose to take it out then?

**Comment by Crystalwarrior** [comment]
...eh? It's just a minor inconsistency fix. It doesn't affect any actual mechanics.

**Comment by Carbonhell** [comment]
Code is ok,is it merge ready @Crystalwarrior

**Comment by Crystalwarrior** [comment]
Yeah pretty sure

**Pull request merged**